addons/account_peppol/models/account_journal.py

```diff
@@ -0,0 +1,22 @@
+# -*- coding:utf-8 -*-
+
+from odoo import api, fields, models
+
+
+class AccountJournal(models.Model):
+    _inherit = 'account.journal'
+
+    account_peppol_proxy_state = fields.Selection(related='company_id.account_peppol_proxy_state')
+    is_peppol_journal = fields.Boolean(string="Account used for Peppol", default=False, compute="_compute_is_peppol_journal", readonly=True)
+
+    def peppol_get_new_documents(self):
+        self.env['account_edi_proxy_client.user'].with_context(target_company_ids=self.company_id.mapped('id'))._cron_peppol_get_new_documents()
```

Comment:
For now, it is only possible to call `peppol_get_new_documents()` from the UI on a single journal. However, this function could potentially be called from several journals, each from a different company. Using the `mapped` here will avoid an error in that case.

Reply:
even on multiple journals with several companies, `self.company_id.ids` works :thinking: also, are we sure we want to allow that behaviour, to call from multiple journals for different companies?

Reply:
I understand your reasoning, but it's a little different in this case
```python
        edi_users = self.env['account_edi_proxy_client.user'].search(
            [('company_id.account_peppol_proxy_state', '=', 'active')])
```
These are the first lines of the `_cron_peppol_get_new_documents()` function.
Later on in the cron, when using the fetched values to populate fields:
```python
            company = edi_user.company_id
```

The cron fetches for all users (from any company) each time it is triggered, so providing company to the model is not useful in this case, you might as well do:
```suggestion
        self.env['account_edi_proxy_client.user']._cron_peppol_get_new_documents()
```


Also, if you want the ids from a recordset, `.ids` will return the same thing as `.mapped('id')`

Reply:
I have added `('company_id', 'in', target_company_ids)` to edi users search domain so that we only search for users in the given companies. Thus, only triggering the computation in the given companies. This was done so that when we click on the `fetch invoice status` on the journal of company A. Only the computation for the edi users in the company A will be triggered and no for all the companies.

Reply:
> even on multiple journals with several companies, `self.company_id.ids` works 🤔 also, are we sure we want to allow that behaviour, to call from multiple journals for different companies?

I imagine that this case could occur when the PO will decide to add a server action on the journal tree view to trigger the computation on several selected journals.